src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -410,6 +410,23 @@ internal static WellKnownMember GetTupleTypeMember(int arity, int position)
         };
 
         /// <summary>
+        /// Returns true if any of the field names aren't the default tuple member name for that position.
+        /// For instance, (int Alice, int Item2) does have friendly names, but (int Item1, int Item2) doesn't.
+        /// </summary>
+        internal bool HasFriendlyNames()
+        {
+            for (var index = 0; index < _fields.Length; index++)
```

Comment:
Would it seem strange that is user specified (int Item1, int Item2) manually we will default to "no names" case?
Perhaps it is easier to just remember in the symbol whether it was provided names or not?